Tests/Shared.Specs/CollectionEquivalencySpecs.cs

```diff
@@ -1598,30 +1616,6 @@ public void When_subject_is_null_and_expectation_is_enumerable_it_should_throw()
                 "Expected subject not to be <null>*");
         }
 
-        [Fact]
-        public void When_the_expectation_is_not_a_multi_dimensional_array_it_should_throw()
```

Comment:
❓ Why is this test no longer valid?

Reply:
Because since v5, the expectation drives the type of comparison. Since it's a string, it'll never trigger the `MultiDimensionalArrayEquivalencyStep`

Reply:
Makes sense. 
Couldn't see how that test should be affected by the other changes in this PR.

Reply:
Don't even understand how this entire bug only surfaced now.